src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/ScreenshotInProcess.cs

```diff
@@ -0,0 +1,593 @@
+
+    private static readonly ReadOnlyMemory<byte> s_pngHeader = new byte[] { 0x89, (byte)'P', (byte)'N', (byte)'G', (byte)'\r', (byte)'\n', 0x1A, (byte)'\n' };
+
+    private static ReadOnlySpan<byte> Ihdr => "IHDR"u8;
+    private static ReadOnlySpan<byte> Idat => "IDAT"u8;
+    private static ReadOnlySpan<byte> Iend => "IEND"u8;
+    private static ReadOnlySpan<byte> Srgb => "sRGB"u8;
+    private static ReadOnlySpan<byte> Gama => "gAMA"u8;
+    private static ReadOnlySpan<byte> Phys => "pHYs"u8;
+    private static ReadOnlySpan<byte> Actl => "acTL"u8;
+    private static ReadOnlySpan<byte> Fctl => "fcTL"u8;
+    private static ReadOnlySpan<byte> Fdat => "fdAT"u8;
+
+    private enum PngColorType : byte
+    {
+        Grayscale = 0,
+        TrueColor = 2,
+        Indexed = 3,
+        GrayscaleWithAlpha = 4,
+        TrueColorWithAlpha = 6,
+    }
+
+    private enum PngCompressionMethod : byte
+    {
+        /// <summary>
+        /// <see href="https://www.w3.org/TR/png/#dfn-deflate">deflate</see> compression with a sliding window of at
+        /// most 32768 bytes.
+        /// </summary>
+        Deflate = 0,
+    }
+
+    private enum PngFilterMethod : byte
+    {
+        /// <summary>
+        /// Adaptive filtering with five basic filter types.
+        /// </summary>
+        Adaptive = 0,
+    }
+
+    private enum PngInterlaceMethod : byte
+    {
+        /// <summary>
+        /// No interlacing.
+        /// </summary>
+        None = 0,
+
+        /// <summary>
+        /// Adam7 interlace.
+        /// </summary>
+        Adam7 = 1,
+    }
+
+    private enum ApngDisposeOp : byte
+    {
+        /// <summary>
+        /// No disposal is done on this frame before rendering the next; the contents of the output buffer are left
+        /// as-is.
+        /// </summary>
+        None = 0,
+
+        /// <summary>
+        /// The frame's region of the output buffer is to be cleared to fully transparent black before rendering the
+        /// next frame.
+        /// </summary>
+        Background = 1,
+
+        /// <summary>
+        /// The frame's region of the output buffer is to be reverted to the previous contents before rendering the next
+        /// frame.
+        /// </summary>
+        Previous = 2,
+    }
+
+    private enum ApngBlendOp : byte
+    {
+        /// <summary>
+        /// All color components of the frame, including alpha, overwrite the current contents of the frame's output
+        /// buffer region.
+        /// </summary>
+        Source = 0,
+
+        /// <summary>
+        /// The frame should be composited onto the output buffer based on its alpha, using a simple OVER operation as
+        /// described in <see href="https://www.w3.org/TR/png/#13Alpha-channel-processing">Alpha Channel Processing</see>.
+        /// </summary>
+        Over = 1,
+    }
+
+    static ScreenshotInProcess()
+    {
+        DataCollectionService.RegisterCustomLogger(
+            static fullPath =>
```

Comment:
What value would be passed as  `fullPath`? If multiple tests fail will they have clear different names in CI?

Reply:
➡️ The full path is determined by the test harness. The test name is part of the full path, so outputs from different tests can generally be identified. There are some exceptions (e.g. theories where inputs change but the test has the same name), but improvements in this space would be owned by microsoft/vs-extension-testing.